bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/metadata/repository/SPIMetadataRepositoryTest.java

```diff
@@ -875,8 +875,8 @@ public void testSPIEquals() throws ProvisionException, URISyntaxException {
 		assertEquals(unit.getTouchpointData().size(), 1);
 		assertEquals(((IRequiredCapability) unit.getRequirements().iterator().next()).getNamespace(), spiRequiredCapability.getNamespace());
 		assertEquals(((IRequiredCapability) unit.getRequirements().iterator().next()).getName(), spiRequiredCapability.getName());
```

Comment:
Why is the cast unnecessary on the next line but not on this one?

Reply:
getName() is not available from IRequirement, but availabe from IRequiredCapability after the cast.

getMin() and getMax() are available from IRequirement without a cast.

Reply:
Thanks, it looked very suspicious from quick look.

Reply:
> Thanks, it looked very suspicious from quick look.

Sure, that's why I used ecj to detect what is unnecessary :)

But always good to double-check.